api-inference-community/api_inference_community/validation.py

```diff
@@ -36,48 +48,69 @@ class Temperature(ConstrainedFloat):
     strict = True
 
 
-class TextGenerationCheck(BaseModel):
-    max_length: Optional[MaxLength] = None
+class CandidateLabels(ConstrainedList):
+    min_items = 1
+    __args__ = [str]
+
+
+class FillMaskParamsCheck(BaseModel):
     top_k: Optional[TopK] = None
-    top_p: Optional[TopP] = None
-    repetition_penalty: Optional[RepetitionPenalty] = None
-    temperature: Optional[Temperature] = None
 
 
-class FillMaskCheck(BaseModel):
+class ZeroShotParamsCheck(BaseModel):
+    candidate_labels: Union[str, CandidateLabels]
+    multi_label: Optional[bool] = None
+
+
+class TextGenerationParamsCheck(BaseModel):
+    max_new_tokens: Optional[MaxLength] = None
```

Comment:
This should be MaxNewTokens which is different from MaxLength imo, max_length has it's purpose if you want to avoid go beyond the model capacity, so it still has some purpose.

Also I'm never a super fan of breaking code symmetry.

Reply:
Fixed.